### PR TITLE
add case verify_dns_name_resolution

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: microsoft/testsuites/core/dns.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class Dns(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case check DNS name resolution by ping bing.com.
+        """,
+        priority=1,
+    )
+    def verify_dns_name_resolution(self, node: Node) -> None:
```
Comment (squirrelsc) [inline microsoft/testsuites/core/dns.py:32]: The test case behavior and name are inconsistent. 
1. The DNS can be tested without any other action, it's worth to have a case. 
2. The unattented-upgrade can be another test case. After upgraded, we can test the dns, reboot and other P0 scenarios.

Comment (LiliDeng) [reply microsoft/testsuites/core/dns.py:32]: update

File: microsoft/testsuites/core/dns.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    This test suite covers DNS name resolution functionality.
+    """,
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class Dns(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case check DNS name resolution by ping bing.com.
+        """,
+        priority=0,
```
Comment (squirrelsc) [inline microsoft/testsuites/core/dns.py]: Make it P1

File: lisa/tools/ping.py [hunk 3]
```diff
@@ -72,6 +76,10 @@ def ping(
                 sudo=True,
             ).wait_result()
         if not ignore_error:
+            if self._dns_name_resolution_issue.findall(result.stdout):
```
Comment (squirrelsc) [inline lisa/tools/ping.py]: It doesn't need to find the special error message, any failure of ping should be reported. Just change the L83 to return the stdout, if it's not too many lines.

File: microsoft/testsuites/core/dns.py (hunk 2 above)
Comment (LiliDeng) [reply microsoft/testsuites/core/dns.py]: update

File: lisa/tools/ping.py (hunk 3 above)
Comment (LiliDeng) [reply lisa/tools/ping.py]: update

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**